x-pack/plugins/security/server/authentication/authenticator.ts

```diff
@@ -105,6 +106,7 @@ const providerMap = new Map<
   [TokenAuthenticationProvider.type, TokenAuthenticationProvider],
   [OIDCAuthenticationProvider.type, OIDCAuthenticationProvider],
   [PKIAuthenticationProvider.type, PKIAuthenticationProvider],
+  [HTTPAuthenticationProvider.type, HTTPAuthenticationProvider],
```

Comment:
Do we get some benefit I'm not realizing from making the `HTTPAuthenticationProvider` part of the `providerMap`? If it wasn't part of the `providerMap`, could we get rid of the custom `validate` function? Separately, I was wondering whether we could pass all of the enabled providers into the `HTTPAuthenticationProvider` constructor, add a method to each provider to denote which auth header should be supported, and use this as the basis for `isSchemeSupported`...

Reply:
> Do we get some benefit I'm not realizing from making the HTTPAuthenticationProvider part of the providerMap?

Not really, was experimenting with different options and just stopped at some point.

> If it wasn't part of the providerMap, could we get rid of the custom validate function? 

Yep, that would simplify the config part.

> Separately, I was wondering whether we could pass all of the enabled providers into the HTTPAuthenticationProvider constructor, add a method to each provider to denote which auth header should be supported, and use this as the basis for isSchemeSupported...

That's a great idea! But I'd propose slightly modified version - `Authenticator` will calculate proper `supportedSchemes` on its own and just give them to `HTTPAuthenticationProvider` so that it doesn't have to deal with other providers or even know that they exist. I'll make that change and you'll tell if it looks good to you or not :)